test/python/circuit/test_controlled_gate.py

```diff
@@ -1595,6 +1599,25 @@ def test_composite(self):
         self.log.info("%s gate count: %d", uqc.name, uqc.size())
         self.assertLessEqual(uqc.size(), 96, f"\n{uqc}")  # this limit could be changed
 
+    def test_mcrz_complexity(self):
+        """Test MCRZ is decomposed using the efficient MC-SU(2) algorithm.
+
+        Regression test of #13427.
+        """
+        basis_gates = ["sx", "x", "rz", "ecr"]
+        pm = generate_preset_pass_manager(
+            optimization_level=3, basis_gates=basis_gates, seed_transpiler=12345
+        )
+
+        num_qubits = 6
+        angle = np.pi / 7
+        qc = QuantumCircuit(num_qubits)
+        qc.append(RZGate(angle).control(num_qubits - 1), qc.qubits)
+
+        isa_qc = pm.run(qc)
+
+        self.assertLessEqual(isa_qc.count_ops().get("ecr", 0), 65)
```

Comment:
Hm maybe it was 65 before the fix, but you're right: it is 40. I'll update!

Reply:
Fixed in ffbb955c3

Reply:
I think the original issues had a CX count of 65, since it was transpiling for the connectivity map of the device, but for all-to-all connectivity the CX-count is lower